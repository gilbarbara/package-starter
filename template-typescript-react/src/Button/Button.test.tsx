import { render, screen } from '@testing-library/react';

import { Button } from './index';

describe('Button', () => {
  it('should render a primary button', () => {
    render(<Button>Primary</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('should render a secondary button', () => {
    render(<Button variant="secondary">Secondary</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
  });
});
