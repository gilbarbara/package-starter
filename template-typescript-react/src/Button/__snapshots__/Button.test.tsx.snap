// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render a primary button 1`] = `
<button
  class="_button_544a72"
  type="button"
>
  Primary
</button>
`;

exports[`Button > should render a secondary button 1`] = `
<button
  class="_button_544a72"
  style="background-color: #ad7bff; color: #000;"
  type="button"
>
  Secondary
</button>
`;
